CONTRIBUTING.md

```diff
@@ -1,39 +1,123 @@
-the end).
+# Contributing to the Firebase JS SDK
+
+We'd love for you to contribute to our source code and to help make the Firebase JS SDK even better than it is today! Here are the guidelines we'd like you to follow:
+
+ - [Code of Conduct](#coc)
+ - [Question or Problem?](#question)
+ - [Issues and Bugs](#issue)
+ - [Feature Requests](#feature)
+ - [Submission Guidelines](#submit)
+ - [Coding Rules](#rules)
+ - [Signing the CLA](#cla)
+
+## <a name="coc"></a> Code of Conduct
+
+As contributors and maintainers of the Firebase JS SDK project, we pledge to respect everyone who contributes by posting issues, updating documentation, submitting pull requests, providing feedback in comments, and any other activities.
+
+Communication through any of Firebase's channels (GitHub, StackOverflow, Google+, Twitter, etc.) must be constructive and never resort to personal attacks, trolling, public or private harassment, insults, or other unprofessional conduct.
+
+We promise to extend courtesy and respect to everyone involved in this project regardless of gender, gender identity, sexual orientation, disability, age, race, ethnicity, religion, or level of experience. We expect anyone contributing to the project to do the same.
+
+If any member of the community violates this code of conduct, the maintainers of the Firebase JS SDK project may take action, removing issues, comments, and PRs or blocking accounts as deemed appropriate.
+
+If you are subject to or witness unacceptable behavior, or have any other concerns, please drop us a line at samstern@google.com. 
+
+
+## <a name="question"></a> Got a Question?
+
+If you have questions about how to use the Firebase JS SDK, please direct these to [StackOverflow][stackoverflow] and use the `firebase` tag. You can also use the [Firebase Google Group][firebase-google-group] or [Slack][slack] to contact members of the Firebase team for help.
+
+## <a name="issue"></a> Found an Issue?
+
+If you find a bug in the source code, a mistake in the documentation, or some other issue, you can help us by submitting an issue to our [GitHub Repository][github]. Even better you can submit a Pull Request with a test demonstrating the bug and a fix!
+
+See [below](#submit) for some guidelines.
+
+## <a name="other-issue"></a> Production Issues
+
+If you have a production issue, please [contact Firebase support][support] who will work with you to resolve the issue. 
+
+## <a name="submit"></a> Submission Guidelines
+
+### Submitting an Issue
+
+Before you submit your issue search [past issues][archive], [StackOverflow][stackoverflow], and the [Firebase Google Group][firebase-google-group] maybe your question was already answered.
+
+If your issue appears to be a bug, and hasn't been reported, open a new issue. Help us to maximize the effort we can spend fixing issues and adding new features, by not reporting duplicate issues. Providing the following information will increase the chances of your issue being dealt with quickly:
```

Comment:
Delete "Help ..." sentence. It was just pointed out in the previous sentence - "Before ...